auditlog/log-enricher/src/main/java/io/debezium/demos/auditing/enricher/TopologyProducer.java

```diff
@@ -0,0 +1,278 @@
+package io.debezium.demos.auditing.enricher;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.json.Json;
+import javax.json.JsonObject;
+
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+public class TopologyProducer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TopologyProducer.class);
+
+    private static final Long BUFFER_OFFSETS_KEY = -1L;
+    private static final String STREAM_BUFFER_NAME = "stream-buffer-state-store";
+    private static final String STORE_NAME = "transaction-meta-data";
+
+
+    @ConfigProperty(name = "audit.context.data.topic")
+    String txContextDataTopic;
+
+    @ConfigProperty(name = "audit.vegetables.topic")
+    String vegetablesTopic;
+
+    @ConfigProperty(name = "audit.vegetables.enriched.topic")
+    String vegetablesEnrichedTopic;
+
+    @Produces
+    public Topology buildTopology() {
+        StreamsBuilder builder = new StreamsBuilder();
+
+        StoreBuilder<KeyValueStore<Long, JsonObject>> streamBufferStateStore =
+                Stores
+                    .keyValueStoreBuilder(
+                        Stores.persistentKeyValueStore(STREAM_BUFFER_NAME),
+                        new Serdes.LongSerde(),
+                        new JsonObjectSerde()
+                    )
+                    .withCachingDisabled();
```

Comment:
@gunnarmorling Why is the caching disabled here?

Reply:
@jpechane, it just seemed safer to me. Tbh. I'm not sure what the notion of "caching" exactly means here, so I decided to go for a better "safe than sorry" approach.

Reply:
@gunnarmorling Caching in this regard means that the changes are applied in memory and into a store backing topic. In case of crash the state is recovered from the state sotrage + changes from the backing topic are replayed. With cachin disabled every change is written directly into state store which can be I/O bottleneck.

Reply:
@jpechane I think you are mixing up two different concepts, namely "caching vs. logging" here w.r.t state stores. both can be configured independently from one another. it's the logging that does the backing based on changelog topics which has in that sense nothing to do with the caching itself which may or may not be used.